ios/RCTMapboxGL/RCTMapboxGLManager.m

```diff
@@ -605,4 +605,85 @@ - (NSArray*)serializePacksArray:(NSArray<MGLOfflinePack*>*)packs
     }];
 }
 
+RCT_EXPORT_METHOD(queryRenderedFeatures:(nonnull NSNumber *)reactTag
+                  options:(NSDictionary *)options
+                  callback:(RCTResponseSenderBlock)callback)
+{
+    [_bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTMapboxGL *> *viewRegistry) {
+        RCTMapboxGL *mapView = viewRegistry[reactTag];
+        if ([mapView isKindOfClass:[RCTMapboxGL class]]) {
+            NSDictionary *pointDict = options[@"point"];
+            NSDictionary *rect = options[@"rect"];
+            if (!pointDict) {
+                // TODO: figure out best way to error this. Should we be using RCTPromise rather than a callback?
+//                reject(@"invalid_arguments", @"queryRenderedFeatures(): screenCoordX and screenCoordY are required.", nil);
+                return;
+            }
+            NSNumber *screenCoordX = pointDict[@"screenCoordX"];
+            NSNumber *screenCoordY = pointDict[@"screenCoordY"];
+            NSSet<NSString *> *styleLayerIdentifiers = options[@"layers"];
```

Comment:
Wouldn’t this be an NSArray rather than an NSSet, given the JavaScript representation? If this works, it’s only because the SDK happens to be calling methods common to both classes. Probably better to convert it into an NSArray using `allObjects`.